modules/dhcp/providers/server/isc.rb

```diff
@@ -149,11 +148,22 @@ def loadSubnetData subnet
 
     def parse_config_for_subnets
       ret_val = []
-
-      @config.scan(/subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)/) do |match|
-        next unless managed_subnet? "#{match[0]}/#{match[1]}"
-
-        ret_val << Proxy::DHCP::Subnet.new(match[0], match[1])
+      @config.scan(/subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)\s*{([\w|;|\.|,|\-|\s]*)}/) do |match|
+        options = {}
+        if match[2]
```

Comment:
Mind to extract this logic to a method or something similar? Maybe write a comment / save `match[0] match[1] match[2]` to named variables? It's a bit difficult to understand without being well acquainted with the format of the conf file.